src/python/pants/init/options_initializer.py

```diff
@@ -162,7 +152,13 @@ def create(cls, options_bootstrapper, build_configuration, init_subsystems=True)
             )
 
         # Parse and register options.
-        options = cls._construct_options(options_bootstrapper, build_configuration)
+
+        known_scope_infos = [
+            si
+            for optionable in build_configuration.all_optionables
+            for si in optionable.known_scope_infos()
+        ]
```

Comment:
You can alternatively use `itertool.chain.from_iterable(optionable.known_scope_infos() for optionable in build_configuration.all_optionables)`. Mostly a style preference.